djvm/src/main/java/sandbox/sun/security/provider/ByteArrayAccess.java

```diff
@@ -0,0 +1,163 @@
+package sandbox.sun.security.provider;
+
+@SuppressWarnings({"WeakerAccess", "unused"})
+final class ByteArrayAccess extends sandbox.java.lang.Object {
+
+    // Convert Little Endian byte[] to int[].
+    @SuppressWarnings("SameParameterValue")
+    static void b2iLittle(byte[] source, int sourceOffset, int[] target, int targetOffset, int sourceSize) {
+        if (sourceOffset < 0 || targetOffset < 0 || sourceOffset + sourceSize > source.length
+                || target.length - targetOffset < (sourceSize / Integer.BYTES)) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+
+        final int sourceEnd = sourceOffset + sourceSize;
+        while (sourceOffset < sourceEnd) {
+            target[targetOffset++] = (
+                (source[++sourceOffset] & 0xFF) |
+                (source[++sourceOffset] & 0xFF) << 8 |
+                (source[++sourceOffset] & 0xFF) << 16 |
+                (source[sourceOffset] & 0xFF) << 24
```

Comment:
In C, the evaluation order of bitwise PR is unspecified. Does the JVM give different guarantees?

Reply:
Apparently, [the evaluation order _is_ specified](https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.7):
> The Java programming language guarantees that the operands of operators appear to be evaluated in a specific evaluation order, namely, from left to right. 

Mind you, it also says not to rely on it...

Reply:
Fixed :+1: